util/webhook/webhook.go

```diff
@@ -244,14 +279,66 @@ func affectedRevisionInfo(payloadIf interface{}) (webURLs []string, revision str
 	return webURLs, revision, change, touchedHead, changedFiles
 }
 
+func fetchDiffstatFromBitbucket(workspace, repoSlug, spec, accessToken string) ([]string, error) {
+	// Getting the files changed from diff API:
+	// https://developer.atlassian.com/cloud/bitbucket/rest/api-group-commits/#api-repositories-workspace-repo-slug-diffstat-spec-get
+
+	url := fmt.Sprintf("https://api.bitbucket.org/2.0/repositories/%s/%s/diffstat/%s", workspace, repoSlug, spec)
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, fmt.Errorf("error creating request: %w", err)
+	}
+
+	trimmedAccessToken := strings.TrimSpace(accessToken)
```

Comment:
Why do we have to trim it? Why can't it be without spaces?

Reply:
Whenever fetching the accessToken from the secrets, a new line is added by default. When this accessToken is passed without removing the spaces/newline, the HTTP GET call fails, with error -  ```invalid header field value for \"Authorization\""```